src/util/minimalStylelint.js

```diff
@@ -0,0 +1,33 @@
+import partial from 'lodash/partial';
+import getPostcssResult from 'stylelint/dist/getPostcssResult';
+import lintSource from 'stylelint/dist/lintSource';
+
+const stylelint = {};
+
+const config = {
+  rules: {
+    'declaration-block-trailing-semicolon': ['always'],
+  },
+  pluginFunctions: {},
+};
+
+Object.assign(stylelint, {
+  _postcssResultCache: new Map(),
+  _options: {},
+
+  getConfigForFile() {
+    return Promise.resolve({config});
+  },
+
+  isPathIgnored() {
+    return Promise.resolve(false);
+  },
+
+  _getPostcssResult: partial(getPostcssResult, stylelint),
+});
+
+export default (code) => {
+  // Prevent postcssResultCache from caching results
```

Comment:
I’m generally against inline comments—if the code doesn’t speak for itself, it can be made clearer! In this case, I think it’s reasonably obvious what’s going on, but if you think it needs further explanation, you could extract a `preventPostcssResultCaching()` function and call it.